lib/ide/src/double_representation/graph.rs

```diff
@@ -3,34 +3,81 @@
 use crate::prelude::*;
 
 use crate::double_representation::definition;
+use crate::double_representation::definition::DefinitionInfo;
+use crate::double_representation::definition::DefinitionName;
+use crate::double_representation::definition::DefinitionProvider;
 use crate::double_representation::node;
 
 use ast::Ast;
+//use ast::HasRepr;
 use ast::known;
 
 
 
+// ================
+// === Graph Id ===
+// ================
+
+/// Crumb describes step that needs to be done when going from context (for graph being a module)
+/// to the target.
+// TODO [mwu]
+//  Currently we support only entering named definitions.
+pub type Crumb = DefinitionName;
+
+/// Identifies graph in the module.
+#[derive(Clone,Debug,Eq,Hash,PartialEq)]
+pub struct Id {
+    /// Sequence of traverses from module root up to the identified graph.
+    pub crumbs : Vec<Crumb>,
+}
```

Comment:
I see this id is cloned pretty often in the API. Im wondering if its should not be made immutable and put behind an Rc to provide a fast clone instead.